dvc/remote/local.py

```diff
@@ -196,9 +190,11 @@ def hardlink(self, from_info, to_info):
     def is_hardlink(path_info):
         return System.is_hardlink(path_info)
 
-    @staticmethod
-    def reflink(from_info, to_info):
-        System.reflink(from_info, to_info)
+    def reflink(self, from_info, to_info):
+        tmp_info = to_info.parent / tmp_fname(to_info.name)
+        System.reflink(from_info, tmp_info)
```

Comment:
Yes, you can change mode on reflink without affecting the source. This is because reflinks have their own inodes, so from fs perspective they are separate files that have their own perms.

Reply:
@efiop, Can you mention this on a code itself as a comment?

Reply:
Also, can this not be done without link to `temp`

Reply:
> @efiop, Can you mention this on a code itself as a comment?

Sure! Done.

> Also, can this not be done without link to temp?

Temp file is needed to make it atomic, so the target file appears already with correct permissions, so there is no chance of something trying to use it while it is in read-only mode.

Reply:
Got it, thanks.